Source/santad/TTYWriter.mm

```diff
@@ -36,20 +37,30 @@
 
 TTYWriter::TTYWriter(dispatch_queue_t q) : q_(q) {}
 
-void TTYWriter::Write(const char *tty, NSString *msg) {
-  dispatch_async(q_, ^{
-    @autoreleasepool {
-      int fd = open(tty, O_WRONLY | O_NOCTTY);
-      if (fd == -1) {
-        LOGW(@"Failed to open TTY for writing: %s", strerror(errno));
-        return;
-      }
+bool TTYWriter::CanWrite(const es_process_t *proc) {
+  return proc && proc->tty && proc->tty->path.length > 0;
+}
+
+void TTYWriter::Write(const es_process_t *proc, NSString *msg) {
+  if (!CanWrite(proc)) {
+    return;
+  }
 
-      std::string_view str = santa::common::NSStringToUTF8StringView(msg);
-      write(fd, str.data(), str.length());
+  // Copy the data from the es_process_t so the ES message doesn't
+  // need to be retained
+  NSString *tty = santa::common::StringToNSString(proc->tty->path.data);
```

Comment:
I needed the lifetime to be extended because it's being used in an async block, so the data would need to be copied out of the `es_process_t`. Putting it in an NSString was mostly convenience since block capturing handles it easily.